gunicorn/reloader.py

```diff
@@ -3,128 +3,15 @@
-                except OSError:
-                    continue
-                old_time = mtimes.get(filename)
-                if old_time is None:
-                    mtimes[filename] = mtime
-                    continue
-                elif mtime > old_time:
-                    if self._callback:
-                        self._callback(filename)
-            time.sleep(self._interval)
-
-has_inotify = False
-if sys.platform.startswith('linux'):
-    try:
-        from inotify.adapters import Inotify
-        import inotify.constants
-        has_inotify = True
-    except ImportError:
-        pass
-
-
-if has_inotify:
-
-    class InotifyReloader(threading.Thread):
-        event_mask = (inotify.constants.IN_CREATE | inotify.constants.IN_DELETE
-                      | inotify.constants.IN_DELETE_SELF | inotify.constants.IN_MODIFY
-                      | inotify.constants.IN_MOVE_SELF | inotify.constants.IN_MOVED_FROM
-                      | inotify.constants.IN_MOVED_TO)
-
-        def __init__(self, extra_files=None, callback=None):
-            super(InotifyReloader, self).__init__()
-            self.setDaemon(True)
-            self._callback = callback
-            self._dirs = set()
-            self._watcher = Inotify()
-
-            for extra_file in extra_files:
-                self.add_extra_file(extra_file)
-
-        def add_extra_file(self, filename):
-            dirname = os.path.dirname(filename)
-
-            if dirname in self._dirs:
-                return
-
-            self._watcher.add_watch(dirname, mask=self.event_mask)
-            self._dirs.add(dirname)
-
-        def get_dirs(self):
-            fnames = [
-                os.path.dirname(re.sub('py[co]$', 'py', module.__file__))
-                for module in list(sys.modules.values())
-                if hasattr(module, '__file__')
-            ]
-
-            return set(fnames)
-
-        def run(self):
-            self._dirs = self.get_dirs()
-
-            for dirname in self._dirs:
-                self._watcher.add_watch(dirname, mask=self.event_mask)
-
-            for event in self._watcher.event_gen():
-                if event is None:
-                    continue
-
-                filename = event[3]
-
-                self._callback(filename)
-
-else:
-
-    class InotifyReloader(object):
-        def __init__(self, callback=None):
-            raise ImportError('You must have the inotify module installed to '
-                              'use the inotify reloader')
-
-
-preferred_reloader = InotifyReloader if has_inotify else Reloader
-
-reloader_engines = {
-    'auto': preferred_reloader,
-    'poll': Reloader,
-    'inotify': InotifyReloader,
-}
+__all__ = ['Reloader', 'reloader_engines']
+
+try:
+    from gunicorn._reloader import Reloader
+except ImportError:
+    class Reloader(object):
+        def __init__(self, *args, **kwargs):
+            raise ImportError('You must have the watchdog module '
```

Comment:
`gunicorn.workers.gaiohttp` raises `RuntimeError`: 

https://github.com/benoitc/gunicorn/blob/0e4e0779e63c080777bfddb02de4cb828cfdb1c7/gunicorn/workers/gaiohttp.py#L12

Should we follow it or raise `ImportError` instead?

Reply:
Or, can we just make watchdog a dependency?

Reply:
I'd like to encourage local development with gunicorn, if possible.

Reply:
I'm not opposed but I'll defer to @benoitc.

Reply:
I will make it a RuntimeError if we can't quickly vendor watchdog.